internal/fs/fs_test.go

```diff
@@ -702,3 +770,86 @@ func TestIsEmpty(t *testing.T) {
 		}
 	}
 }
+
+func TestIsSymlink(t *testing.T) {
+	dir, err := ioutil.TempDir("", "dep")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	dirPath := filepath.Join(dir, "directory")
+	if err = os.MkdirAll(dirPath, 0777); err != nil {
+		t.Fatal(err)
+	}
+
+	filePath := filepath.Join(dir, "file")
+	f, err := os.Create(filePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	f.Close()
+
+	dirSymlink := filepath.Join(dir, "dirSymlink")
+	fileSymlink := filepath.Join(dir, "fileSymlink")
+	if err = os.Symlink(dirPath, dirSymlink); err != nil {
+		t.Fatal(err)
+	}
+	if err = os.Symlink(filePath, fileSymlink); err != nil {
+		t.Fatal(err)
+	}
+
+	var (
+		inaccessibleFile    string
+		inaccessibleSymlink string
+	)
+
+	err, cleanup := setupInaccessibleDir(func(dir string) error {
+		inaccessibleFile = filepath.Join(dir, "file")
+		if fh, err := os.Create(inaccessibleFile); err != nil {
+			return err
+		} else if err = fh.Close(); err != nil {
+			return err
+		}
+
+		inaccessibleSymlink = filepath.Join(dir, "symlink")
+		err = os.Symlink(inaccessibleFile, inaccessibleSymlink)
+		return err
+	})
+	defer cleanup()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := map[string]struct {
+		expected bool
+		err      bool
+	}{
+		dirPath:             {false, false},
+		filePath:            {false, false},
+		dirSymlink:          {true, false},
+		fileSymlink:         {true, false},
+		inaccessibleFile:    {false, true},
+		inaccessibleSymlink: {false, true},
+	}
+
+	for path, want := range tests {
+		if runtime.GOOS == "windows" {
+			// XXX: setting permissions works differently in
+			// Microsoft Windows. Skipping this this until a
+			// compatible implementation is provided.
+			t.Skip("skipping on windows")
+		}
+
+		got, err := IsSymlink(path)
+		if err != nil {
+			if !want.err {
+				t.Fatalf("expected no error, got %v", err)
```

Comment:
use t.Errorf(); continue

so that you get all the test failures, not just the first